adagrams/game.py

```diff
@@ -1,11 +1,152 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
 def draw_letters():
-    pass
+    used_letters = {}
+    letters = []
+    possible_letters = list(LETTER_POOL.keys())
+
+    while len(letters) <= 9:
+        random_int = random.randint(0, 25)
+        random_l =  possible_letters[random_int]
+        if not used_letters.get(random_l) == LETTER_POOL[random_l]:
+            if (not used_letters.get(random_l)):
+                used_letters[random_l] = 0 
+            used_letters[random_l] += 1
+            letters.append(random_l)
+        
+    return letters
+
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    upper_word = word.upper()
+    letter_bank_dict = {}
+    is_valid = False
+
+    for character in upper_word:
+        if character in letter_bank:
+            is_valid = True
+        else:
+            is_valid = False
+            return is_valid
+    
+
+    if character in letter_bank_dict.keys():
+        letter_bank_dict[character] += 1
+    else:
+        letter_bank_dict[character] = 1
+
+    character_frequency_word = upper_word.count(character)
+    character_frequency_dict = letter_bank_dict[character]
+
+    if character_frequency_word <= character_frequency_dict:
+        return True
+    else:
+        if character not in letter_bank or character_frequency_word >= character_frequency_dict:
+            return False
+
 
 def score_word(word):
-    pass
+    upper_word = word.upper()
+    total_points = 0
+    points_dict = {
+    'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10 }
```

Comment:
Optionally, we could declutter this function by moving this dictionary outside and declaring it as a constant variable.